src/uam/workers/sensing_data/ar00_worker.cpp

```diff
@@ -0,0 +1,138 @@
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include <uam/protocol_types/uam_protocol_types.h>
+#include <uam/workers/sensing_data/ar00_worker.h>
+
+#include <optional>
+#include <string_view>
+
+namespace uam
+{
+AR00Worker::AR00Worker(const uint32_t idx_offset) :
+  WorkerBase<AR00Worker, 'A', 'R', '0', '0', protocol::AR00CommandReply>(
+    idx_offset,
+    idx_offset + offsetof(Reply, footer)),
+  sensing_data_visitor_(idx_offset + protocol::c_sensing_data_start_idx),
+  distances_visitor_(idx_offset + protocol::c_distance_start_idx)
+{
+}
+
+std::optional<AR00Worker::Reply> AR00Worker::decode(const std::string_view& buffer) const
+{
+  Reply reply;
+  if (!decodeHeaderAndFooter(buffer, reply))
+    return std::nullopt;
+  if (!decodeSensingData(buffer, reply.sensing_data))
+    return std::nullopt;
+  // if (!decodeDistances(buffer, reply.ranges))
+  //   return std::nullopt;
+  return reply;
+}
+
+std::optional<AR00Worker::Reply> AR00Worker::decode(const Reply& raw_reply) const
+{
+  Reply reply = raw_reply;
+  if (!decodeHeaderAndFooter(reply))
+    return std::nullopt;
+  if (!BaseType::decodeSensingData(reply.sensing_data))
+    return std::nullopt;
+  // Decode ranges
+  if (!decodeField(reply.ranges))
+    return std::nullopt;
+  return reply;
+}
+
+bool AR00Worker::decodeSensingData(
+  const std::string_view& buffer,
+  protocol::sensing_data::SensingDataHeader& sensing_data) const
+{
+  if (!sensing_data_visitor_.operating_mode.get(buffer, sensing_data.operating_mode))
+    return false;
+  if (!sensing_data_visitor_.area_number.get(buffer, sensing_data.area_number))
+    return false;
+
+  // Grab the Error Status
+  if (!sensing_data_visitor_.error_state.get(buffer, sensing_data.error_state))
+    return false;
+  // Grab the error code and offset by 0x40 if non-zero as per documentation
+  if (!sensing_data_visitor_.error_code.get(buffer, sensing_data.error_code))
+    return false;
+  if (sensing_data.error_code != 0)
+  {
+    sensing_data.error_code += 0x40;
```

Comment:
This error code `0x40` is in several places. Couldn't this be defined as a constant? For easier maintenance

Reply:
[Duplicated] I've just seen the comment below